CHANGELOG.md

```diff
@@ -10,6 +10,9 @@ IMPROVEMENTS:
  * env: Default interpolation of optional meta fields of parameterized jobs to
    an empty string rather than the field key. [[GH-3720](https://github.com/hashicorp/nomad/issues/3720)]
 
+BUG FIXES:
+ * driver/docker: Fix docker credential helper support [[GH-3818](https://github.com/hashicorp/nomad/issues/3818)] [[GH-4221](https://github.com/hashicorp/nomad/issues/4221)]
```

Comment:
Can we say what case is fixed? I am assuming it worked for some subset of image specification formats

Reply:
I think it was a fluke that it worked for gcr as the implementation didn’t follow the current spec. Maybe it changed along the way? I don’t think it’s used much outside of GCE.

Reply:
I think it was a fluke that it worked for gcr as the implementation didn’t follow the current spec. Maybe it changed along the way? I don’t think it’s used much outside of GCE.

Reply:
Okay. Also we should like to only GH-4266. This PR should mark those two issues as closed.

Reply:
Yeah it looks like the aws helper's [regex](https://github.com/awslabs/amazon-ecr-credential-helper/blob/master/ecr-login/api/client.go#L44) doesn't care whats after the hostname.

[GCR is using URL.Parse](https://github.com/GoogleCloudPlatform/docker-credential-gcr/blob/master/credhelper/helper.go#L252) and [verifying the Host is a part of GCR](https://github.com/GoogleCloudPlatform/docker-credential-gcr/blob/master/credhelper/helper.go#L112).

Reply:
the gcloud helper which i just found out is actually an alias into the google cloud sdk [expects the url](https://github.com/google-cloud-sdk/google-cloud-sdk/blob/master/lib/surface/auth/docker_helper.py#L58) to be [one of these](https://github.com/google-cloud-sdk/google-cloud-sdk/blob/master/lib/googlecloudsdk/core/docker/constants.py#L37)